projects/packages/publicize/src/class-publicize-base.php

```diff
@@ -1128,6 +1129,25 @@ public function register_post_meta() {
 			register_meta( 'post', $this->POST_DONE . 'all', $already_shared_flag_args );
 			register_meta( 'post', self::POST_JETPACK_SOCIAL_OPTIONS, $jetpack_social_options_args );
 		}
+
+		add_filter(
+			'default_post_metadata',
+			function ( $value, $object_id, $meta_key ) {
+				if ( $meta_key === '_wpas_options' ) {
+					$post = get_post( $object_id );
+					if (
+						! empty( $post->post_status ) &&
+						'auto-draft' !== $post->post_status &&
+						! empty( $value[0]['image_generator_settings']['enabled'] )
```

Comment:
I don't quite understand why the `value` array has a `0` index, like there would be multiple default values, but this does work.

The other thing to check is if a post has a value for `_wpas_options` (because they've shared an image) will we still check the defaults for `image_generator_settings`. If not then we should probably flip the logic, so the default is `false` but we set it to `true` for `auto-draft` posts.

Reply:
Ok, I've realised what's going on here now. When `get_metadata` is called with `$single=false` then the value is expected to be an array, and so we get passed the default value as an array of length 1.

I've reworked this function a bit to cope with both values for `$single` as, between the tests, and the default loading of a post in the block editor, it can be set differently.